### PR TITLE
ArC - document io.quarkus.arc.WithCaching

### LINKED ISSUE #17625: ArC - document io.quarkus.arc.WithCaching
This annotation can be used together with `javax.enterprise.inject.Instance` to cache the result of the `Instance#get()` operation even for `@Dependent` beans.

### PR DESCRIPTION
**Pull request by mkouba**
- resolves #17625

### CODE REVIEW

File: docs/src/main/asciidoc/cdi-reference.adoc [hunk 1]
```diff
@@ -784,6 +784,65 @@ class NaiveLoggingInterceptor {
...
+     
+   @Dependent
+   @Produces 
+   Long produceLong() {
+     return nextLong.incrementAndGet();
+   }
+}
+
+import io.quarkus.arc.WithCaching;
+import javax.enterprise.inject.Instance;
```
Comment (Ladicek) [inline docs/src/main/asciidoc/cdi-reference.adoc:819]: Maybe remove the imports here? It looks weird, in the middle of a continuous block of code.

File: docs/src/main/asciidoc/cdi-reference.adoc [hunk 2]
```diff
@@ -784,6 +784,65 @@ class NaiveLoggingInterceptor {
...
+  
+class Consumer {
+  
+  @Inject
+  Instance<Long> longInstance;
+
+  @WithCaching
+  @Inject
```
Comment (Ladicek) [inline docs/src/main/asciidoc/cdi-reference.adoc:827]: I personally always write `@Inject` first and qualifiers after that, but that's just my preference :-)

**Review by Ladicek — APPROVED** [review]

File: docs/src/main/asciidoc/cdi-reference.adoc (hunk 2 above)
Comment (mkouba) [reply docs/src/main/asciidoc/cdi-reference.adoc:827]: `WithCaching` is not a qualifier ;-). But you're right that `@Inject @WithCaching` reads much better!

Comment (Ladicek) [reply docs/src/main/asciidoc/cdi-reference.adoc:827]: > `WithCaching` is not a qualifier ;-)

It certainly looks like one! :-)

**Review by manovotn — APPROVED** [review]

**Review by Ladicek — APPROVED** [review]

**Pull request merged**